src/frontend/github-client-middleware.ts

```diff
@@ -6,23 +6,20 @@ import Logger from "bunyan";
 import { booleanFlag, BooleanFlags } from "../config/feature-flags";
 
 export default (octokitApp: App): RequestHandler => async (req: Request, res: Response, next: NextFunction): Promise<void> => {
-	if (req.session.githubToken) {
-		res.locals.github = GithubAPI({
-			auth: req.session.githubToken
-		});
-	} else {
-		res.locals.github = GithubAPI();
-	}
+	const { jiraHost } = res.locals;
+
+	// If githubToken isn't set, this GithubAPI will be unauthed
```

Comment:
Hmm.... we are not passing it to GithubAPI below, does it mean it is always unauthed, or it will be set up later somewhere?

Reply:
@mboudreau ?

Reply:
`locals.github` is unauthed until we hit the `checkGithubOauth` function which validates the token and sets a new `locals.github` with the token.  If it's not a valid token, gets a new one from github.

This whole way of setting the github client in the locals is not great and needs to be changed, and this was a workaround to centralize the token validation.